src/ea_rest_template.py

```diff
@@ -15,8 +15,8 @@ def ea_rest_call(endpoint: str, methodtype: str, token: str, params=None, data=N
 
     response = requests.request(methodtype, url, headers=headers, params=params, data=json.dumps(data))
 
-    if response.status_code == 200:
-        data = response.json()
-        return data
-    else:
-        print(f"Error reaching API at {url}: \nCode: {response.status_code} \nResponse: {response.headers}")
+    if response.status_code != 200:
+        response.raise_for_status()
```

Comment:
```suggestion
    response.raise_for_status()
```
raise_for_status checks the status code itself and raises exception if its between 400 and 600